### PR TITLE
Gateway address script update

### PR DESCRIPTION
**Pull request by DonnaRyanMicrosoft**
This update adds support for GCCH, CIDR Notated Subnets, and retrieves the data in a much easier fashion. I have had some customers and coworkers test the script and it is functioning as expected.

### CODE REVIEW

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 [hunk 1]
```diff
@@ -98,26 +115,76 @@ function invoke-download {
         $count = $count + 1
         $index = $subnet.IndexOf('/')
         if ($count -ne $subnets.count) {
-        ($subnet.Substring(0, $index)) + ',' | out-file $PSSCriptRoot\temp.csv -Append
+            if ($IP -eq $true) { ($subnet.Substring(0, $index)) + ',' | out-file $PSSCriptRoot\temp.csv -Append }
```
Comment (AshleyYangSZ) [inline Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:118]: What's the value of IP? Is it a string or Boolean? If boolean, IMO, "isIP" or something like that could indicate is a boolean would be better name.

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 [hunk 2]
```diff
@@ -98,26 +115,76 @@ function invoke-download {
...
+            { $subnet | out-file $PSSCriptRoot\temp.csv -Append }
         }
     }    
     write-host "$count gateways found"
+
+
 }
 
 #Removes left over temp.csv file if found
+if (($GOV -eq $true) -and ($Azure -eq $true)){write-host "Using Azure as the source is not supported with GCCH. Remove the -Azure parameter and try again."
```
Comment (AshleyYangSZ) [inline Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:166]: Same comment for GOV parameter

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 [hunk 3]
```diff
@@ -98,26 +115,76 @@ function invoke-download {
...
 }
 
-invoke-modulecheck
+#Removes left over temp.json file if found
+if (Test-Path -Path $PSScriptRoot\temp.json) {
+    Remove-Item -Path $PSScriptRoot\temp.json -Force -ErrorAction Stop
+}
 
-connect-azure
+if ($Azure -eq $true) {
```
Comment (AshleyYangSZ) [inline Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:179]: Same ask for this $Azure parameter.

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 (hunk 1 above)
Comment (DonnaRyanMicrosoft) [reply Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:118]: It is a boolean. All of the flagged variables are boolean. I see your point when looking at this from a code "readability" perspective, but I think using "isIP" as a variable name may be confusing to users who actually need to call these variables at command line. In this scenario, I think the existing names make sense.

Thoughts?

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 (hunk 2 above)
Comment (DonnaRyanMicrosoft) [reply Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:166]: Commented in first issue.

File: Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1 (hunk 3 above)
Comment (DonnaRyanMicrosoft) [reply Windows 365 Gateway IP Lookup/Get-AVDW365Gateways.ps1:179]: Commented in first issue

**Review by AshleyYangSZ — APPROVED** [review]

**Pull request merged**